### DISCUSSION TITLE
Is it possible to change the font-size property at runtime while using EmbedResourcesKind::EmbedForSoftwareRenderer

### DISCUSSION THREAD

**Discussion by MiSimon** [slint-ui/slint#4940] [Q&A] [answered]
Hello,

i want to be able to change the font-size property at runtime, to do this i use a global property that contains a scale factor value and this value is multiplied with the default font-size.

```
in property <float> font-scale-factor: Globals.scale-factor;
// ...

out property <Typo> typos: {
    label_small: {
        font-family: root.default-text-font-family,
        font-size: 12px * root.font-scale-factor,
        font-weight: 400,
    },
    //...
}

export component LabelBase inherits Text {
    in property <FontStyle> font;

    font-family: font.font-family;
    font-size: font.font-size;
    font-weight: font.font-weight;

    color: Theme.brushes.foreground;
    
    vertical-stretch: 0;
    horizontal-stretch: 0;
}

export component LabelSmall inherits LabelBase { 
    font: Theme.typos.label-small;
}
```

This works in the preview but not if i compile and run it.
Applying the scale factor to geometry properties like padding and spacing works in both cases.

build.rs:
```
fn main() {
    slint_build::compile_with_config(
        "ui/main.slint",
        slint_build::CompilerConfiguration::new()
            .with_style("fluent-light".to_owned())
            .embed_resources(slint_build::EmbedResourcesKind::EmbedForSoftwareRenderer),
    )
    .unwrap();    
}
```

Cargo.toml:
```
[dependencies.slint]
version = "1.5.0"
default-features = false
features = ["compat-1-2", "unsafe-single-threaded", "libm", "log"]
```

Are dynamic font sizes supported with this configuration?

Best regards,
Michael

**Comment by ogoffart** [comment, marked as answer]
We're missing a way to configure font properly.
Right now this can be done by setting the `SLINT_FONT_SIZES` env variable when building
For example: `SLINT_FONT_SIZES=8,11,10,12,13,14,15,16,18,20,22,24,32`
(i realize this is not documented)

**Reply by MiSimon** [reply]
Thank you for the response. I tried this but it leads to a panic in the build command:

`thread 'main' panicked at 'called Result::unwrap() on an Err value: CompileError(["S:\\src\\common\\gui\\gui\\ui/main.slint:20: Invalid font size '8 10 11 12 13 14 15 16 18 20 22 24 32' specified in SLINT_FONT_SIZES"])', S:\\src\common\gui\gui\build.rs:8:6`

**Reply by ogoffart** [reply]
They need to be separated by comas in the env variable

**Reply by MiSimon** [reply]
I forgot the double quotes around them. Now it works, thank you!
Are you planning to support this in a later version, it seems that this solution has significant impact on the resulting file size.

**Reply by ogoffart** [reply]
Yes, this is a planed change. Filled https://github.com/slint-ui/slint/issues/4956
